test/powershell/Host/PSVersionTable.Tests.ps1

```diff
@@ -22,4 +24,22 @@ Describe "PSVersionTable" -Tags "CI" {
     It "Should have the correct edition" -Skip:(!$IsCoreCLR) {
 	$PSVersionTable["PSEdition"] | Should Be "Core"
     }
+
+    It "Should have the correct platform info" {
+    $platform = [String][System.Environment]::OSVersion.Platform
+	[String]$PSVersionTable["Platform"] | Should Be $platform
+    }
+
+    It "Should have the correct OS info" {
+    if ($IsCoreCLR)
+    {
+        $OSDescription = [String][System.Runtime.InteropServices.RuntimeInformation]::OSDescription
+	    [String]$PSVersionTable["OS"] | Should Be $OSDescription
+    }
+    else
+    {
+        $OSDescription = [String][System.Environment]::OSVersion
+        [String]$PSVersionTable["OS"] | Should Be [System.Environment]::OSVersion
```

Comment:
Part of me thinks we shouldn't make any changes for FullCLR, but if we're making the change it seems this line would fail as the right hand side should use ToString()